### PR TITLE
Feature/597 docs and images in playground params

### PR DESCRIPTION
**Pull request by neoxelox**
Related to: https://github.com/latitude-dev/latitude-llm/issues/597

### CODE REVIEW

File: apps/infra/src/app/production/shared.ts [hunk 1]
```diff
@@ -150,6 +150,10 @@ export const environment = pulumi
       { name: 'DRIVE_DISK', value: 's3' },
       { name: 'ASW_REGION', value: 'eu-central-1' },
       { name: 'S3_BUCKET', value: 'latitude-llm-bucket-production' },
+      {
+        name: 'PUBLIC_S3_BUCKET',
+        value: 'latitude-llm-public-bucket-production',
+      },
```
Comment (geclos) [inline apps/infra/src/app/production/shared.ts:156]: I'd recommend moving the infra changes to another PR and merge that PR beforehand. Also you'll need to manually deploy workers and websockets in order for this envvar to be available there (gateway and web are automatically updated when you deploy)

File: apps/infra/src/core/s3.ts [hunk 2]
```diff
@@ -18,3 +18,18 @@ export const bucket = new aws.s3.BucketV2(
...
+    bucket: 'latitude-llm-public-bucket-production',
+    tags: {
+      Name: 'Latitude LLM public bucket',
+      Environment: 'Production',
+    },
+  },
+  { provider: regionProvider },
+)
+
+export const publicBucketName = publicBucket.bucket
```
Comment (geclos) [inline apps/infra/src/core/s3.ts:35]: looks good

File: .gitignore [hunk 3]
```diff
@@ -45,6 +45,10 @@ next-env.d.ts
 apps/web/scripts-dist
 
 # File uploads in development
+apps/web/uploads
```
Comment (geclos) [inline .gitignore:48]: hmm why two upload folders

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/Input/index.tsx [hunk 4]
```diff
@@ -72,3 +95,232 @@ export function InputParams({
...
+          defaultFilename={input.metadata.filename}
+          onChange={onImageChange}
+          accept={SUPPORTED_IMAGE_TYPES.join(',')}
+          multiple={false}
+          disabled={disabled}
+        />
+      )
+
+    default:
+      return null
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/Input/index.tsx:324]: should we throw an error or is this an expected case?

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx [hunk 5]
```diff
@@ -163,23 +167,33 @@ function DocumentLogParameters({
...
-      {} as Inputs<'manual'>,
-    )
-  }, [documentLog.uuid, documentLog.parameters])
-
   return (
     <>
       <Text.H5M color='foreground'>Parameters</Text.H5M>
-      <InputParams inputs={parameters} disabled />
+      <div className='grid grid-cols-[auto_1fr] gap-y-3'>
+        {Object.entries(documentLog.parameters).map(
```
Comment (geclos) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx:174]: could documentlog parameters be null or undefined?

File: apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.test.ts [hunk 6]
```diff
@@ -6,17 +8,33 @@ describe('recalculateInputs', () => {
...
+          parameters: {
+            param1: {
+              type: ParameterType.Text,
+            },
+            param2: {
+              type: ParameterType.Document,
+            },
+          },
+        },
+      } as unknown as ConversationMetadata,
```
Comment (geclos) [inline apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.test.ts:29]: why do you need this unknown escape hatch?

File: apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.ts [hunk 7]
```diff
@@ -1,32 +1,57 @@
...
+        return [
+          param,
+          {
+            ...input,
+            metadata: {
+              ...input.metadata,
+              includeInPrompt: input.metadata.includeInPrompt ?? true,
+              ...config[param],
+              type: type,
+            },
```
Comment (geclos) [inline apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.ts:39]: although smart this is complicated to parse for the reader, i have basically no idea what this method is returning. Would it be simpler to just write the object keys and assign them to the appropiate values?

File: apps/web/src/stores/files.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  )
+
+  return {
+    uploadFile,
+    convertFile,
+    isUploadingFile,
+    isConvertingFile,
+    isLoading: isUploadingFile || isConvertingFile,
+  }
+}
```
Comment (geclos) [inline apps/web/src/stores/files.ts:94]: fantastic

File: docs/guides/prompt-manager/playground.mdx [hunk 9]
```diff
@@ -37,20 +37,64 @@ When parameters are present in your prompt, they will automatically appear in th
...
 
 Load data from a dataset to populate the parameters, where each row in the dataset corresponds to a set of values for the parameters.
 ![Dataset Parameters](/assets/parameters_dataset.png)
 [Learn more about using datasets in Latitude](/guides/datasets/overview)
 
-### History
+#### History
 
 Load values from previous runs of the prompt to populate the parameters.
 ![History Parameters](/assets/parameters_history.png)
```
Comment (geclos) [inline docs/guides/prompt-manager/playground.mdx:100]: beautiful

File: packages/core/src/constants.ts [hunk 10]
```diff
@@ -469,3 +469,11 @@ export const RELATIVE_DATES = {
 } as const
 
 export type RelativeDate = keyof typeof RELATIVE_DATES
+
+export enum ParameterType {
+  Text = 'text',
+  Document = 'document',
```
Comment (geclos) [inline packages/core/src/constants.ts:475]: nitpick: File is shorter and matches with the wording OAI uses for it's type file message

File: packages/core/src/lib/disk.ts [hunk 11]
```diff
@@ -49,20 +57,23 @@ async function getReadableStreamFromFile(file: File) {
 export class DiskWrapper {
   private disk: Disk
 
-  constructor() {
-    this.disk = new Disk(this.buildDisk())
+  constructor(visibility: 'private' | 'public' = 'private') {
+    this.disk = new Disk(this.buildDisk(visibility))
```
Comment (geclos) [inline packages/core/src/lib/disk.ts:61]: tbh this class would probably benefit from a refactor, it should receive an instance of Disk as a parameter an otherwise default to the local disk instance. Maybe leave a TODO?

File: packages/web-ui/src/ds/atoms/DropzoneInput/index.tsx [hunk 12]
```diff
@@ -18,27 +26,40 @@ export function DropzoneInput({
...
+
+  const onClearFile = useCallback(() => {
+    if (ref.current) ref.current.value = ''
+    setFilename(undefined)
+    onChange?.(null)
+  }, [ref, setFilename, onChange])
+
   return (
-    <div onClick={onClickZone}>
+    <div onClick={onClickZone} className='w-full'>
```
Comment (geclos) [inline packages/web-ui/src/ds/atoms/DropzoneInput/index.tsx:62]: hmm review this ain't breaking UI in other places where this component is used

**Review by geclos — APPROVED** [review]
lovely work only not would be to try to split PRs in smaller chunks for the future. An example split would be:
- A PR implementing the server actions/services
- A PR implementing the frontend plumbing

I know I have a 14k monster PR pending but that was a failure on my side.

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/Input/index.tsx (hunk 4 above)
Comment (neoxelox) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/Input/index.tsx:320]: we could throw an error

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx (hunk 5 above)
Comment (neoxelox) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/DocumentLogs/DocumentLogInfo/Metadata.tsx:174]: In theory no, but I will check for that just in case

File: apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.ts (hunk 7 above)
Comment (neoxelox) [reply apps/web/src/hooks/useDocumentParameters/recalculateInputs/index.ts:39]: okay

File: packages/core/src/constants.ts (hunk 10 above)
Comment (neoxelox) [reply packages/core/src/constants.ts:475]: mmm, the thing is that we try to convert the document to plain text. If we say 'file' everywhere it seems that I can upload an image or even a video...

Comment (neoxelox) [reply packages/core/src/constants.ts:475]: Also, the new content type from Anthropic does refer to it as document too:
```
"content": [{
            "type": "document",
            "source": {
                "type": "base64",
                "media_type": "application/pdf",
                "data": $PDF_BASE64
            }
        },
```

File: packages/web-ui/src/ds/atoms/DropzoneInput/index.tsx (hunk 12 above)
Comment (neoxelox) [reply packages/web-ui/src/ds/atoms/DropzoneInput/index.tsx:62]: Reviewed, it does not break current uses 👍

File: packages/core/src/lib/disk.ts (hunk 11 above)
Comment (neoxelox) [reply packages/core/src/lib/disk.ts:61]: okay

File: apps/web/src/hooks/useDocumentParameters/index.ts [hunk 13]
```diff
@@ -113,6 +119,7 @@ export function useDocumentParameters({
   documentVersionUuid: string
   commitVersionUuid: string
 }) {
+  // TODO: Delete stale inputs as new inputs could eventually not fit
```
Comment (andresgutgon) [inline apps/web/src/hooks/useDocumentParameters/index.ts:122]: What means this?
